core/modules/system/system.install

```diff
@@ -602,6 +600,43 @@ function system_requirements($phase) {
     }
   }
 
+  // Warn the user if trusted hostnames have not been configured.
+  if ($phase == 'runtime') {
+    $trusted_host_patterns = settings_get('trusted_host_patterns', array());
+    $more_info = $t('See <a href="@url" target="_blank">Protecting against HTTP HOST Header attacks</a> for more information.', array('@url' => 'https://api.backdropcms.org/documentation/trusted-host-settings'));
+    // @todo: upgrade this to an error when we provide a way to configure
+    // $trusted_host_patterns via the admin UI.
+    if (empty($trusted_host_patterns)) {
+      $requirements['trusted_host_patterns'] = array(
+        'title' => $t('Trusted Host Setting'),
+        'value' => $t('Not configured'),
+        'description' => $t('The <code>trusted_host_patterns</code> setting is not configured in <code>settings.php</code>. It is highly recommended that you configure this to protect your site.') . ' ' . $more_info,
+        'severity' => $trusted_host_patterns === FALSE ? REQUIREMENT_INFO : REQUIREMENT_WARNING,
```

Comment:
This didn't seem to be doing anything during my testing. I always saw a warning when there was no pattern match - not any info message ...if this is meant for the installer, then we need `if ($phase == 'runtime' || $phase == 'install') {`. Right?

Reply:
...oh well, maybe this works. The `backdrop` recipe lando is using skips the requirements steps (unless there are errors) 🤔

Reply:
Per the updated documentation in settings.php:
```
 * If you do not need this functionality (such as in development environments or
 * if protection is at another layer), you can suppress the status report
 * warning by setting this value to FALSE:
 * @code
 * $settings['trusted_host_patterns'] = FALSE;
 * @endcode
 *
```

So you have to set it explicitly to FALSE. If you don't define it at all, then you get the warning.